Installation issue: tix

### Steps to reproduce the issue

```console
$ spack install tix
...
```


### Error message

<details><summary>Error message</summary>

<pre>
...
==> tix: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'


</pre></details>


### Information on your system

* **Spack:** 0.23.0.dev0 (17c02fe7592cc6a4c2afe51f879c52cdcb3aa549)
* **Python:** 3.9.12
* **Platform:** linux-rhel8-broadwell
* **Concretizer:** clingo

### Additional information

==> tix: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

32 errors found in build log:
     113    /usr/WS2/lawrence31/spack/lib/spack/env/gcc/gcc -DPACKAGE_NAME=\"Tix\" -DPACKAGE_TARNAME=\"tix\" -DPACKAGE_VERSION
            =\"8.4.3\" -DPACKAGE_STRING=\"Tix\ 8.4.3\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_S
            YS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_ST
            DINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SA
            FE=1 -DTCL_THREADS=1 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_IS_LONG=1 -DUSE_TCL_STUBS=1 -DUSE_TK_STUBS=1   -I. -I"
            /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic" -I. -I"/
            var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/unix" -I"/usr/WS2/
            lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/share/tcl/
            src/generic" -I"/usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk
            5fx2kwbqrqpkii/share/tcl/src/unix" -I"/usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tk-8.6.
            11-kbzbzjtpkc3midkczkp5girai2to332l/share/tk/src/generic" -I"/usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broad
            well/gcc-10.3.1/tk-8.6.11-kbzbzjtpkc3midkczkp5girai2to332l/share/tk/src/unix" -I/usr/WS2/lawrence31/spack/opt/spac
            k/linux-rhel8-broadwell/gcc-10.3.1/libx11-1.8.9-lwzx64olna7wkwharusrk7uge2unw7dh/include    -pipe -O2 -fomit-frame
            -pointer -Wall -Wno-implicit-int -fPIC  -c `echo /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ew
            qjbehnjfqsswyh6zplaci/spack-src/generic/tixDiImg.c`
     114    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     115                     from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tk-8.6.11-kbzbzjtpkc3mi
            dkczkp5girai2to332l/share/tk/src/generic/tk.h:19,
     116                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tix.h:30,
     117                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixInt.h:19,
     118                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixScroll.c:11:
  >> 119    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     120     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     121          |              ^~
     122    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     123     2630 | #   define panic  Tcl_Panic
     124          |                   ^~~~~~~~~
     125    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     126       63 | EXTERN void panic(char * format, ...);
     127          |             ^~~~~
  >> 128    make: *** [Makefile:293: tixScroll.o] Error 1
     129    make: *** Waiting for unfinished jobs....
     130    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     131                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     132                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixGeometry.c:16:
  >> 133    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     134     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     135          |              ^~
     136    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     137     2630 | #   define panic  Tcl_Panic
     138          |                   ^~~~~~~~~
     139    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     140       63 | EXTERN void panic(char * format, ...);
     141          |             ^~~~~
     142    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     143                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     144                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixDiITxt.c:16:
  >> 145    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     146     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     147          |              ^~
     148    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     149     2630 | #   define panic  Tcl_Panic
     150          |                   ^~~~~~~~~
     151    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     152       63 | EXTERN void panic(char * format, ...);
     153          |             ^~~~~
     154    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     155                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixUtils.c:16:
  >> 156    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     157     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     158          |              ^~
     159    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     160     2630 | #   define panic  Tcl_Panic
     161          |                   ^~~~~~~~~
     162    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     163       63 | EXTERN void panic(char * format, ...);
     164          |             ^~~~~
     165    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     166                     from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tk-8.6.11-kbzbzjtpkc3mi
            dkczkp5girai2to332l/share/tk/src/generic/tk.h:19,
     167                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixOption.c:25:
  >> 168    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     169     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     170          |              ^~
     171    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     172     2630 | #   define panic  Tcl_Panic
     173          |                   ^~~~~~~~~
     174    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     175       63 | EXTERN void panic(char * format, ...);
     176          |             ^~~~~
     177    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     178                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     179                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixError.c:16:
  >> 180    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     181     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     182          |              ^~
     183    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     184     2630 | #   define panic  Tcl_Panic
     185          |                   ^~~~~~~~~
     186    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     187       63 | EXTERN void panic(char * format, ...);
     188          |             ^~~~~
     189    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     190                     from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tk-8.6.11-kbzbzjtpkc3mi
            dkczkp5girai2to332l/share/tk/src/generic/tk.h:19,
     191                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixClass.c:28:
  >> 192    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     193     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     194          |              ^~
     195    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     196     2630 | #   define panic  Tcl_Panic
     197          |                   ^~~~~~~~~
     198    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     199       63 | EXTERN void panic(char * format, ...);
     200          |             ^~~~~
  >> 201    make: *** [Makefile:293: tixGeometry.o] Error 1
     202    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixUtils.
            c:674:22: warning: initialization of 'const char * (*)(void *, struct Tk_Window_ *, char *, int,  void (**)(char *
            ))' from incompatible pointer type 'char * (*)(void *, struct Tk_Window_ *, char *, int,  void (**)(char *))' [-Wi
            ncompatible-pointer-types]
     203      674 |     ReliefParseProc, ReliefPrintProc, 0,
     204          |                      ^~~~~~~~~~~~~~~
     205    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixUtils.
            c:674:22: note: (near initialization for 'tixConfigRelief.printProc')
     206    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     207                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     208                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixSmpLs.c:17:
  >> 209    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     210     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     211          |              ^~
     212    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     213     2630 | #   define panic  Tcl_Panic
     214          |                   ^~~~~~~~~
     215    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     216       63 | EXTERN void panic(char * format, ...);
     217          |             ^~~~~
     218    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     219                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     220                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixList.c:15:
  >> 221    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     222     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     223          |              ^~
     224    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     225     2630 | #   define panic  Tcl_Panic
     226          |                   ^~~~~~~~~
     227    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     228       63 | EXTERN void panic(char * format, ...);
     229          |             ^~~~~
     230    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     231                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     232                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixInit.c:18:
  >> 233    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     234     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     235          |              ^~
     236    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     237     2630 | #   define panic  Tcl_Panic
     238          |                   ^~~~~~~~~
     239    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     240       63 | EXTERN void panic(char * format, ...);
     241          |             ^~~~~
  >> 242    make: *** [Makefile:293: tixSmpLs.o] Error 1
  >> 243    make: *** [Makefile:293: tixError.o] Error 1
  >> 244    make: *** [Makefile:293: tixList.o] Error 1
  >> 245    make: *** [Makefile:293: tixUtils.o] Error 1
  >> 246    make: *** [Makefile:293: tixOption.o] Error 1
  >> 247    make: *** [Makefile:293: tixInit.o] Error 1
     248    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     249                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     250                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixDItem.c:25:
  >> 251    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     252     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     253          |              ^~
     254    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     255     2630 | #   define panic  Tcl_Panic
     256          |                   ^~~~~~~~~
     257    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     258       63 | EXTERN void panic(char * format, ...);
     259          |             ^~~~~
  >> 260    make: *** [Makefile:293: tixDiITxt.o] Error 1
  >> 261    make: *** [Makefile:293: tixClass.o] Error 1
     262    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixDItem.
            c: In function 'Tix_WidgetConfigure2':
     263    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixDItem.
            c:448:19: warning: variable 'widgRecList' set but not used [-Wunused-but-set-variable]
     264      448 |     CONST84 char *widgRecList[2];
     265          |                   ^~~~~~~~~~~
     266    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixDItem.
            c: At top level:
     267    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixDItem.
            c:517:21: warning: initialization of 'const char * (*)(void *, struct Tk_Window_ *, char *, int,  void (**)(char *
            ))' from incompatible pointer type 'char * (*)(void *, struct Tk_Window_ *, char *, int,  void (**)(char *))' [-Wi
            ncompatible-pointer-types]
     268      517 |     DItemParseProc, DItemPrintProc, 0,
     269          |                     ^~~~~~~~~~~~~~
     270    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixDItem.
            c:517:21: note: (near initialization for 'tixConfigItemType.printProc')
     271    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     272                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     273                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixCompat.c:16:
  >> 274    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     275     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     276          |              ^~
     277    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     278     2630 | #   define panic  Tcl_Panic
     279          |                   ^~~~~~~~~
     280    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     281       63 | EXTERN void panic(char * format, ...);
     282          |             ^~~~~
     283    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     284                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     285                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixCmds.c:17:
  >> 286    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     287     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     288          |              ^~
     289    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     290     2630 | #   define panic  Tcl_Panic
     291          |                   ^~~~~~~~~
     292    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     293       63 | EXTERN void panic(char * format, ...);
     294          |             ^~~~~
     295    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     296                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixPort.h:22,
     297                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixDiImg.c:16:
  >> 298    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     299     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     300          |              ^~
     301    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     302     2630 | #   define panic  Tcl_Panic
     303          |                   ^~~~~~~~~
     304    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     305       63 | EXTERN void panic(char * format, ...);
     306          |             ^~~~~
  >> 307    make: *** [Makefile:293: tixCompat.o] Error 1
     308    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     309                     from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tk-8.6.11-kbzbzjtpkc3mi
            dkczkp5girai2to332l/share/tk/src/generic/tk.h:19,
     310                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixMethod.c:17:
  >> 311    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     312     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     313          |              ^~
     314    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     315     2630 | #   define panic  Tcl_Panic
     316          |                   ^~~~~~~~~
     317    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     318       63 | EXTERN void panic(char * format, ...);
     319          |             ^~~~~
  >> 320    make: *** [Makefile:293: tixDItem.o] Error 1
  >> 321    make: *** [Makefile:293: tixDiImg.o] Error 1
  >> 322    make: *** [Makefile:293: tixMethod.o] Error 1
  >> 323    make: *** [Makefile:293: tixCmds.o] Error 1
     324    In file included from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tcl.h:2441,
     325                     from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tclPort.h:25,
     326                     from /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicp
            cwzzdk5fx2kwbqrqpkii/share/tcl/src/generic/tclInt.h:36,
     327                     from /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack
            -src/generic/tixWidget.c:15:
  >> 328    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     329     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
     330          |              ^~
     331    /usr/WS2/lawrence31/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/tcl-8.6.11-rwfsmjtzmicpcwzzdk5fx2kwbqrqpkii/s
            hare/tcl/src/generic/tcl.h:2630:19: note: in expansion of macro 'Tcl_Panic'
     332     2630 | #   define panic  Tcl_Panic
     333          |                   ^~~~~~~~~
     334    /var/tmp/lawrence31/spack-stage/spack-stage-tix-8.4.3-swfa7kdy6ewqjbehnjfqsswyh6zplaci/spack-src/generic/tixPort.h
            :63:13: note: in expansion of macro 'panic'
     335       63 | EXTERN void panic(char * format, ...);
     336          |             ^~~~~
  >> 337    make: *** [Makefile:293: tixWidget.o] Error 1

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/tix/package.py
+++ b/var/spack/repos/builtin/packages/tix/package.py
@@ -29,10 +29,15 @@ class Tix(AutotoolsPackage):
         sha256="1be1a1c7453f6ab8771f90d7e7c0f8959490104752a16a8755bbb7287a841a96",
         level=0,
     )
+    # This patch causes 'install' to fail on RHEL8 with at least gcc@10.3.1 with
+    # "error: expected ')' before '->' token in expansion of macro 'Tcl_Panic'".
+    #
+    # TBD: Is the problem the platform, os, or compiler?
     patch(
         "https://raw.githubusercontent.com/macports/macports-ports/v2.7.0-archive/x11/tix/files/implicit.patch",
         sha256="8a2720368c7757896814684147029d8318b9aa3b0914b3f37dd5e8a8603a61d3",
         level=0,
+        when="platform=darwin",
     )
     patch(
         "https://raw.githubusercontent.com/macports/macports-ports/v2.7.0-archive/x11/tix/files/patch-generic-tixGrSort.c.diff",

